comp110-worksheet-7/DirectoryUtils.cs

```diff
@@ -24,37 +34,70 @@ public static bool IsDirectory(string path)
 		// Return the total size, in bytes, of all the files below the given directory
 		public static long GetTotalSize(string directory)
 		{
-			throw new NotImplementedException();
+			valuesReference = new ImportantValues();
+			void sizeRead() //reads the size of each file in a directory
+			{
+				string[] files = new string[0];
+				files = Directory.GetFiles(directory); //Assigns each file in the directory to an array
+				foreach (string file in files)
+				{
+					valuesReference.fileSize += new FileInfo(file).Length;
+
+					valuesReference.fileDirectories[0] = valuesReference.fileDirectories[0] + file;
+					valuesReference.allSizes[0] = valuesReference.allSizes[0] + valuesReference.fileSize;
+					valuesReference.fileNo = valuesReference.fileNo + 1;
+					if(IsDirectory(file) != true)
+					{
+						valuesReference.fileDepth = valuesReference.fileDepth + 1;
+						Recursion(files[0], file);
+					}
+				}
+			}
+			void Recursion(string files, string file)
```

Comment:
Does this need to be a separate function? I think it would be more readable if you just had a single recursive function rather than two mutually-recursive ones.